### PR TITLE
SONP-1173 add pro[erties to use maven plugin for Azure DevOps Server

### PR DESCRIPTION
**Pull request by AndrejLavrinovic**
* Configuration properties are added for Azure DevOps Server plugin.
* Some explanations and details are added into README file
* Tests added to verify new functionality

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -212,13 +214,17 @@ App password permissions required are:
...
     See Microsoft documentation here for details on creating a Personal Access Token. Please ensure that the token has at minimum the scope Code (read and write) for the organization you wish to access
 
 * A list of `AZURE_DEVOPS_SERVICES` organizations that you currently have access to can be found [here](https://aex.dev.azure.com/me). Also listed are the projects for each organization.
 
-    Enter just the organization, without ‘dev.azure.com’ at the beginning or ‘.visualstudio.com’ at the end. For example, if your organization is ‘dev.azure.com/example-user’ or 'example-user.visualstudio.com’, enter just 'example-user’. 
+    Enter just the organization, without ‘dev.azure.com’ at the beginning or ‘.visualstudio.com’ at the end. For example, if your organization is ‘dev.azure.com/example-user’ or 'example-user.visualstudio.com’, enter just 'example-user’.
+
+* `AZURE_DEVOPS_SERVER` is a local instance of Azure DevOps services, thus corresponding url must be provided e.g. http://locahost:3434
+
+* `AZURE_DEVOPS_SERVER` contains ProjectCollection property which must be included into configurations with its default value - `DefaultCollection`
```
Comment (michaelmelodysmartbear) [inline README.md]: You can remove the above 2 lines, anyone familiar with TFS/DevOps Server would be able be knowledgeable enough to understand these properties in conjunction with the above table of properties.

File: README.md [hunk 2]
```diff
@@ -170,7 +170,9 @@ Parameter | Description | Required? | SCM Specific? | Default
 **`repository`** | The repository to push SwaggerHub changes to | yes | - | - 
 **`repositoryOwner`** | The SCM account which owns the above repository | no | `GITHUB`. `BITBUCKET` | - 
 **`scmProject`** | Team Project which contains the target repository | no | `AZURE_DEVOPS_SERVICES` | - 
-**`scmOrganization`** | Specific to Azure DevOps Services, the organization with which to synchronize | no | `AZURE_DEVOPS_SERVICES` | - 
+**`scmOrganization`** | Specific to Azure DevOps Services, the organization with which to synchronize | no | `AZURE_DEVOPS_SERVICES` | -
+**`scmUrl`** | Specific to Azure DevOps Server, Instance Url | no | `AZURE_DEVOPS_SERVER` | -
```
Comment (michaelmelodysmartbear) [inline README.md]: Can shorten this to `host URL of the SCM`

File: README.md [hunk 3]
```diff
@@ -170,7 +170,9 @@ Parameter | Description | Required? | SCM Specific? | Default
...
+**`scmProjectCollection`** | Project Collection which contains the Team Project associated with API | no | `AZURE_DEVOPS_SERVER` | DefaultCollection
```
Comment (michaelmelodysmartbear) [inline README.md]: CAn shorten to `Project collection which contains the target repositories project.`

File: src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubClient.java [hunk 4]
```diff
@@ -112,7 +112,7 @@ private Request buildGetRequest(HttpUrl httpUrl, MediaType mediaType) {
             }
             return Optional.ofNullable(response);
         } catch (IOException e) {
-            log.error(String.format("Error when attempting to save % plugin integration for API %s. Error message %s", saveSCMPluginConfigRequest.getScmProvider(), saveSCMPluginConfigRequest.getApi(), e.getMessage()));
+            log.error(String.format("Error when attempting to save %s plugin integration for API %s. Error message %s", saveSCMPluginConfigRequest.getScmProvider(), saveSCMPluginConfigRequest.getApi(), e.getMessage()));
```
Comment (michaelmelodysmartbear) [inline src/main/java/io/swagger/swaggerhub/plugin/SwaggerHubClient.java]: Good stop 👍

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java [hunk 5]
```diff
@@ -156,7 +158,9 @@ private SwaggerHubClient buildSwaggerHubClient(String basePath){
                 .scmPassword(SCM_PASSWORD)
                 .project(SCM_PROJECT)
                 .account(SCM_ACCOUNT)
-                .personalAccessToken(SCM_PERSONAL_ACCESS_TOKEN);
+                .personalAccessToken(SCM_PERSONAL_ACCESS_TOKEN)
+                .url(SCM_TFS_URL)
+                .projectCollection(SCM_TFS_PROJECT_COLLECTION);
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: Remove references to TFS in the constant property names. We may have other SCM integrations in future which use those same properties

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java [hunk 6]
```diff
@@ -176,7 +180,9 @@ private RequestPatternBuilder putRequestPattern(String url){
                 .withRequestBody(matchingJsonPath("$.password", equalTo(SCM_PASSWORD)))
                 .withRequestBody(matchingJsonPath("$.account", equalTo(SCM_ACCOUNT)))
                 .withRequestBody(matchingJsonPath("$.project", equalTo(SCM_PROJECT)))
-                .withRequestBody(matchingJsonPath("$.personalAccessToken", equalTo(SCM_PERSONAL_ACCESS_TOKEN)));
+                .withRequestBody(matchingJsonPath("$.personalAccessToken", equalTo(SCM_PERSONAL_ACCESS_TOKEN)))
+                .withRequestBody(matchingJsonPath("$.url", equalTo(SCM_TFS_URL)))
+                .withRequestBody(matchingJsonPath("$.projectCollection", equalTo(SCM_TFS_PROJECT_COLLECTION)));
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubClientTest.java]: Again, remove references to `TFS`

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java [hunk 7]
```diff
@@ -471,6 +530,17 @@ private RequestPatternBuilder createPutSCMConfigRequestPatternWithAccountPATandP
 
     }
 
+    private RequestPatternBuilder createPutSCMConfigRequestPatternWithUrlAndProjectCollection(String owner, String api,String version, String oasVersion, String branch,
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java]: Really small point, might be overkill - instead within this method you could `return createPutSCMConfigRequestPatternWithAccountPATandProject` and chain `.withRequestBody(matchingJsonPath("$.projectCollection"...` on there also. Would shorten the method a little.

Again it's no biggie.

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java [hunk 8]
```diff
@@ -294,6 +294,65 @@ public void testMultiDefinitionsAreUploaded_andSCMSaveRequestMadeWithAccountPATA
...
+        verify(saveSCMPluginRequestPattern3);
+    }
+
+    @Test
+    public void testInputFileIsUploaded_andSCMSaveRequestMadeWithUrlAndProjectCollection() throws Exception {
+        //GIVEN
+        UrlPathPattern saveDefinitionRequest = stubSaveDefinitionRequest(API_OWNER, INPUT_FILE_API, INPUT_FILE_API_VERSION, IS_PRIVATE, OAS2, JSON, SWAGGERHUB_API_TOKEN);
+        UrlPathPattern saveSCMPluginConfigurationRequest = stubSaveSCMPluginConfigurationRequest(API_OWNER, INPUT_FILE_API, INPUT_FILE_API_VERSION, OAS2, SWAGGERHUB_API_TOKEN);
+
+        //Add test for output file value
```
Comment (michaelmelodysmartbear) [inline src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java]: Remove comment

File: src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java (hunk 7 above)
Comment (AndrejLavrinovic) [reply src/test/java/io/swagger/swaggerhub/plugin/SwaggerHubUploadTest.java]: @michaelmelodysmartbear -- 
Would be better to do that way you are mentioned above, but Azure DevOps Server doesn't use `$.account` property which is configured in `createPutSCMConfigRequestPatternWithAccountPATandProject` method. In this case we'll need to add
```
<scmOrganization>account</scmOrganization>
```
property to the plugin configuration that is not what we need for TFS configs.
